### ISSUE TITLE
No matching cache files found

### ISSUE THREAD

**Issue by CheezusCrisp** [PhantomGamers/SteamFriendsPatcher#9]
[24-Jun-20 16:52:10] [Info] Checking for updates...
[24-Jun-20 16:52:10] [Info] Force scan started.
[24-Jun-20 16:52:10] [Info] Checking for latest friends.css...
[24-Jun-20 16:52:11] [Info] No updates found.
[24-Jun-20 16:52:12] [Info] Successfully downloaded latest friends.css
[24-Jun-20 16:52:12] [Info] Finding list of possible cache files...
[24-Jun-20 16:52:12] [Error] No matching cache files found.
[24-Jun-20 16:52:12] [Error] Please confirm that Steam is running and that the friends list is open and try again.
[24-Jun-20 16:52:12] [Info] Cache Watcher Started.

**Comment by PhantomGamers** [comment]
For some reason Valve messed up the way the cache files are stored which breaks this version of the patcher for some users. For now use this for the friends skin https://github.com/PhantomGamers/EnableNewSteamFriendsSkin

**Comment by CheezusCrisp** [comment]
So this issue is only on some people I assume? Will there be an update for the friends patcher to support the different cache location?

**Comment by PhantomGamers** [comment]
Yes it's only some people. It's not a different cache location, it's the same location but a mismatch in the file size. The reason the old patcher works is because it decompresses every file and checks against the raw css file, I can add this functionality into the new patcher as an option but it would be potentially more problematic with the auto scanner as it would use more resources... and this is definitely a bug because the file size is still correct in the cache's index file so I'm hoping maybe it just gets fixed by Valve/Google (don't know if it's a valve bug or a chromium bug).

**Comment by Thespikedballofdoom** [comment]
Getting this on the new version. Wasn't having this issue before v 0.1.25 beta.

`[06/08/20 7:46:57 AM] [Error] No matching cache files found.
[06/08/20 7:46:57 AM] [Error] Please confirm that Steam is running and that the friends list is open and try again.`

OLD VERSION
`[06/08/20 7:52:41 AM] [Info] Finding list of possible cache files...
[06/08/20 7:52:41 AM] [Info] Found 2 possible cache files.
[06/08/20 7:52:41 AM] [Info] Checking cache files for match...
[06/08/20 7:52:41 AM] [Info] Successfully found matching friends.css at C:\Users\itsab\AppData\Local\Steam\htmlcache\Cache\f_000d01.
[06/08/20 7:52:41 AM] [Info] Overwriting with patched version...
[06/08/20 7:52:41 AM] [Info] Reloading friends window...`

**Comment by Scunkaneli** [comment]
> Getting this on the new version. Wasn't having this issue before v 0.1.25 beta.

Same here, broken on v.0.1.25-beta. Downgrading to v0.1.23-beta fixes the issue :)

**Comment by PhantomGamers** [comment]
Are you sure it works consistently on 0.1.23-beta and never on 0.1.25-beta? Is it possible that it just happened the first time you tried 0.1.23-beta out of luck?

The issue I'm currently working on is related on something on valve's end which should affect all versions of this patcher and the cli patcher the same, basically 0.1.25-beta will just additionally find a 3rd version of the css which the older versions will miss. But the reason it's failing (for most people) is because sometimes the patcher will get a different version than the one your client gets. Force Checking (multiple times) *should* fix it but I'm noticing that for some reason force check always searches for the first css file the patcher downloads, so in reality you currently have to restart the patcher in order to get a fresh search... it's very annoying.

**Comment by PhantomGamers** [comment]
I found an issue where actually sometimes it would download the wrong file and be less consistent than previous versions, this might be related to your problems. Fixed in 0.1.26-beta

**Comment by Scunkaneli** [comment]
> Are you sure it works consistently on 0.1.23-beta and never on 0.1.25-beta? Is it possible that it just happened the first time you tried 0.1.23-beta out of luck?

Yeah I could've just been lucky. I was using 0.1.23-beta since release without problems. Then I did the update to 0.1.25-beta and I try to get the new version working, but even after restarting steam for the 10th time it still uses the old friends UI. I downgrade back to 0.1.23-beta and I get the new friends UI on my first try and it persists through multiple steam restarts.

Anyways, I've successfully used 0.1.26-beta for three times in a row now, so it seems that my issue was fixed. Thank you :)
